testing/helpers.py

```diff
@@ -0,0 +1,70 @@
+import http
+import json
+import typing
+import urllib.request
+
+from docker_push_latest_if_changed import _inspect_image
+
+
+class Image(typing.NamedTuple):
+    tag: str
+    name: str
+    registry_uri: str
+    name_tag: str
+    registry_tag: str
+
+
+def get_image(name, tag, registry_uri):
+    if tag:
+        name_tag = f'{name}:{tag}'
+        registry_tag = f'{registry_uri}/{name}:{tag}'
+    else:
+        name_tag = name
+        registry_tag = f'{registry_uri}/{name}'
+    return Image(
+        tag=tag,
+        name=name,
+        registry_uri=registry_uri,
+        name_tag=name_tag,
+        registry_tag=registry_tag,
+    )
+
+
+def is_image_on_registry(image):
+    manifest = None
+    try:
+        manifest = _get_manifest(image)
+    except urllib.error.HTTPError as e:
+        if e.getcode() == http.HTTPStatus.NOT_FOUND.value:
+            pass
+        else:
+            raise
+    return True if manifest else False
```

Comment:
https://github.com/asottile/scratch/wiki/PythonHintsForNoobs#useless-branches

I'd write this:

```python
try:
    _get_manifest(image)
except ...:
    if ...:
        return False
    else:
        raise
else:
    return True
```

Reply:
I was trying to keep from having multiple return statements, but forcing that does make it real derpy.
Changed to as suggested.